WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -43,20 +43,21 @@ public async Task ConfirmConnectionAsync(TimeSpan confirmInterval, CancellationT
 
 		private async Task<bool> ConfirmConnectionAsync()
 		{
-			var inputWeight = 4 * Constants.P2wpkhInputVirtualSize;
+			var inputWeight = Constants.WitnessScaleFactor * Constants.P2wpkhInputVirtualSize;
 			var inputRemainingWeights = new[] { (long)ArenaClient.ProtocolMaxWeightPerAlice - Coins.Count() * inputWeight };
 
 			var amountCredentials = ArenaClient.AmountCredentialClient.Credentials;
 
 			var totalFeeToPay = FeeRate.GetFee(Coins.Sum(c => c.ScriptPubKey.EstimateInputVsize()));
 			var totalAmount = Coins.Sum(coin => coin.Amount);
+			var effectiveAmmout = totalAmount - totalFeeToPay;
 
-			if (totalFeeToPay > totalAmount)
+			if (effectiveAmmout <= Money.Zero)
```

Comment:
Shouldn't this be `effectiveAmmout < Money.Zero`?

Reply:
The coordinator will reject any registration lees than `MinRegistrableAmount` and then here we should check that value but we don't have it yet so, if we already know `0` is not valid then it makes no sense to allow it here. Another question could be what's the point to register an input which amount is exactly the same than the fee required to pay the fee?

Reply:
#5534 added an error in the round state, and #5531 exposes it so that economical inputs are rejected at input registration, so the alice client shouldn't be able to make it this

Reply:
I don't understand what you say. I know this is correct but is it redundant? should I remove it?

Reply:
oops, I distinctly remember writing a reply to this but it appears I never posted it...

strictly speaking it is redundant, since connection confirmation can only happen after successful input registration.

i don't mind keeping it or removing it, redundant checks like that are hard to exercise in test coverage, personally i sometimes find them helpful for readability and sometimes i find them confusing because they contain dead code...


having this check on the client side makes complete sense, but i think it would need to be in whatever class ends up creating the alice client, before it tries to do the input registration that would lead to the alice client being created, so this this check is not just correct, i think it's important, but the right place for it just doesn't exist yet